consensus/spos/worker.go

```diff
@@ -58,70 +61,62 @@ type Worker struct {
 	mutReceivedHeadersHandler sync.RWMutex
 
 	antifloodHandler consensus.P2PAntifloodHandler
+	poolAdder        PoolAdder
+}
+
+// WorkerArgs holds the consensus worker arguments
+type WorkerArgs struct {
+	ConsensusService         ConsensusService
+	BlockChain               data.ChainHandler
+	BlockProcessor           process.BlockProcessor
+	Bootstrapper             process.Bootstrapper
+	BroadcastMessenger       consensus.BroadcastMessenger
+	ConsensusState           *ConsensusState
+	ForkDetector             process.ForkDetector
+	KeyGenerator             crypto.KeyGenerator
+	Marshalizer              marshal.Marshalizer
+	Hasher                   hashing.Hasher
+	Rounder                  consensus.Rounder
+	ShardCoordinator         sharding.Coordinator
+	SingleSigner             crypto.SingleSigner
+	SyncTimer                ntp.SyncTimer
+	HeaderSigVerifier        RandSeedVerifier
+	ChainID                  []byte
+	NetworkShardingCollector consensus.NetworkShardingCollector
+	AntifloodHandler         consensus.P2PAntifloodHandler
+	PoolAdder                PoolAdder
 }
 
 // NewWorker creates a new Worker object
-func NewWorker(
-	consensusService ConsensusService,
-	blockChain data.ChainHandler,
-	blockProcessor process.BlockProcessor,
-	bootstrapper process.Bootstrapper,
-	broadcastMessenger consensus.BroadcastMessenger,
-	consensusState *ConsensusState,
-	forkDetector process.ForkDetector,
-	keyGenerator crypto.KeyGenerator,
-	marshalizer marshal.Marshalizer,
-	rounder consensus.Rounder,
-	shardCoordinator sharding.Coordinator,
-	singleSigner crypto.SingleSigner,
-	syncTimer ntp.SyncTimer,
-	headerSigVerifier RandSeedVerifier,
-	chainID []byte,
-	networkShardingCollector consensus.NetworkShardingCollector,
-	antifloodHandler consensus.P2PAntifloodHandler,
-) (*Worker, error) {
+func NewWorker(args *WorkerArgs) (*Worker, error) {
```

Comment:
If we pass the argument as value, the check from line 144 becomes useless (worker is instantiated once, no performance impact)